src/core/net/srp_client.cpp

```diff
@@ -721,6 +730,15 @@ void Client::SendUpdate(void)
 
     VerifyOrExit(message != nullptr, error = kErrorNoBufs);
     SuccessOrExit(error = PrepareUpdateMessage(*message));
+
+    if (message->GetLength() + sizeof(Ip6::Udp::Header) + sizeof(Ip6::Header) >= Ip6::kMaxDatagramLength)
+    {
+        LogInfo("Msg is larger than MTU, enabling single service mode");
+        mSingleServiceMode.Enable();
+        IgnoreError(message->SetLength(0));
+        SuccessOrExit(error = PrepareUpdateMessage(*message));
```

Comment:
What if we exceed the MTU again even in single service mode?

Reply:
It would be same behavior as the code today, i.e. service registration will fail/timeout and callback is invoked informing user.

We can consider this a user error (i.e., if user adds a single service that won't fit in an MTU). 
In general, would prefer to keep OT SRP client implementation lean and small since it is intended to run on all types of devices including resource-limited and computationally less-capable ones like simple SED sensors. So we try to avoid adding extra checks for incorrect use (bad parameters, etc).